src/CMakeLists.txt

```diff
@@ -48,46 +48,135 @@ endif()
+
+if(NOT JEMALLOC_LIBRARIES)
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(JEMALLOC jemalloc) # Add REQUIRED if we get rid of the fall-back (see below if not JEMALLOC_FOUND).
+  if(JEMALLOC_FOUND)
+    if((NOT JEMALLOC_LIBRARIES) OR (NOT JEMALLOC_INCLUDEDIR))
       message(FATAL_ERROR
-                "Could not find jemalloc library ([${lib_name}]) location via CMake uber-searcher "
-                "find_library().  Do not fret: Please build a jemalloc if not yet done and try again; "
+                "CMake pkg-config searcher claims JEMALLOC_FOUND but has not set "
+                "JEMALLOC_LIBRARIES [${JEMALLOC_LIBRARIES}] and/or JEMALLOC_INCLUDEDIR [${JEMALLOC_INCLUDEDIR}].  "
+                "Do not fret: Please build a jemalloc if not yet done and try again; "
                 "if it is still not found then please supply the full path to the library file (including "
-                "the file itself) as the CMake knob (cache setting) JEMALLOC_LIB.")
+                "the file itself) as the CMake knob (cache setting) JEMALLOC_LIBRARIES; and the full path "
+                "to the include root (the directory containing, but not equal to, jemalloc/) as the CMake knob "
+                "(cache setting) JEMALLOC_INCLUDEDIR.")
     endif()
-    message(VERBOSE "CMake uber-searcher find_library() located jemalloc lib (see below); "
-                      "though you can override this via cache setting.")
-  endblock()
-endif()
+    message(VERBOSE "CMake pkg-config searcher located jemalloc lib + include-dir (see below); "
+                      "though you can override this via cache settings.")
+  else()
+    # Commenting this out for now:
+    #   message(FATAL_ERROR
+    #             "CMake pkg-config searcher could not find jemalloc: JEMALLOC_FOUND not set.  "
+    #             "Do not fret: Please build a jemalloc if not yet done and try again; "
+    #             "if it is still not found then please supply the full path to the library file (including "
+    #             "the file itself) as the CMake knob (cache setting) JEMALLOC_LIBRARIES; and the full path "
+    #             "to the include root (the directory containing, but not equal to, jemalloc/) as the CMake knob "
+    #             "(cache setting) JEMALLOC_INCLUDEDIR.")
+
+    # Here we should be just failing: either via message(FATAL_ERROR) (commented-out) or just by using `REQUIRED`
+    # in the pkg_check_modules() call above.  Instead we do what FlowLikeLib.cmake doc header specifically recommends
+    # against: we use an extra fallback (in the form of find_library() and find_path()).  Why the exception?  Answer:
+    # We know of a case where much of jemalloc is installed (the lib, headers) but ancillary items are not,
+    # namely (as relevant here) the .pc file and (as relevant when determining FLOW_IPC_JEMALLOC_PREFIX below)
+    # the jemalloc-config binary.  That case is when using the current (as of 2/2025) jemalloc Conan recipe;
```

Comment:
dates with only numeric values are so inconsistent worldwide, I would always write `(as of February 2025)` here.